src/common/lib/command_mode.c

```
@@ -170,462 +168,602 @@ static uint16_t brightness_rainbow_hue = 0;      /**< Current rainbow hue (0-359
+}
+
+/**
+ * @brief Process keyboard input in COMMAND_ACTIVE state
+ *
+ * Handles command key presses (B, D, F, L, S) and transitions to appropriate states.
+ * User can release shifts once command mode is active.
+ *
+ * @param keyboard_report Pointer to keyboard HID report
+ * @return false to suppress all keyboard reports
+ */
+static bool process_command_active(const hid_keyboard_report_t* keyboard_report) {
```

Clang (14.0.6)

[warning] function 'process_command_active' has cognitive complexity of 36 (threshold 25)

(readability-function-cognitive-complexity)